raiden_contracts/data/source/raiden/Distribution.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity 0.6.10;
+
+import "test/CustomToken.sol";
+import "raiden/Utils.sol";
+import "lib/ECVerify.sol";
+
+contract Distribution is Utils {
+    CustomToken public token;
+
+    // Chain ID as specified by EIP155 used in balance proof signatures to
+    // avoid replay attacks
+    uint256 public chain_id;
+
+    constructor(
+        address _token_address,
+        uint256 _chain_id
+    )
+        public
+    {
+        require(_token_address != address(0x0));
+        require(_chain_id > 0);
+
+        token = CustomToken(_token_address);
+        chain_id = _chain_id;
+    }
+
+    function claim(
+        address owner,
+        address partner,
+        uint256 amount,
+        bytes memory signature
+    ) public {
+        address receiver = recoverAddressFromClaim(
+            owner,
+            partner,
+            amount,
+            signature
+        );
+
+        require(receiver == owner, "Signature mismatch");
+        token.mintFor(amount, receiver);
```

Comment:
Why should the claim be signed by the owner?

Reply:
The claim should not mint for a address rather than directly depositing it into a channel shouldnt it?

Reply:
This is all WIP, I just started with the most simple way. But we need a distributor address.

Reply:
true

Reply:
So you mint for any address which is then going to deposit it into channels, right?

Reply:
I think for settlement we need to further evolve this. Basically we need to be able to split a claim over two participants.